mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -763,6 +761,13 @@ struct BlockwiseReduceRewritePattern
     }
   }
 
+  int64_t roundToNextPowerof2(int64_t n) const {
```

Comment:
Please check the LLVM support libraries for this function, it should exist.

Also, this might be incorrect once you're in the range where `(double)(i) == (double)(i + 1)`?